RandLAPACK/comps/rl_util.hh

```diff
@@ -370,6 +370,189 @@ void gen_exp_mat(
     }
 }
 
+/// Generates matrix with a staircase spectrum with 4 steps
+template <typename T>
+void gen_step_mat(
+    int64_t& m,
+    int64_t& n,
+    std::vector<T>& A,
+    int64_t k,
+    T cond,
+    bool diagon,
+    RandBLAS::base::RNGState<r123::Philox4x32> state
```

Comment:
Don't template with a specific RNGState type. Start by having ``template <typename T, typename RNG>`` before the function name. Then, for the ``state`` argument, declare it as type ``RandBLAS::base::RNGState<RNG>`` rather than ``RandBLAS::base::RNGState<r123::Philox4x32>``.

Please make this change throughout ``rl_util.hh`` as appropriate. Once you do this, please make sure you don't include Random123 in this file.

Reply:
@TeachRaccooon I'm calling attention to this comment. You'll need fixes throughout this file.

Reply:
Per slack concersation: wil fix later.